### PR TITLE
Restructuring comments to JSDOC

### PR DESCRIPTION
**Pull request by z3dev**
First of many changes for the move to JSDOC

And a little more information added to the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
-# csg.js
+## csg.js
 
+## Constructive Solid Geometry (CSG) Library
 
 [![GitHub version](https://badge.fury.io/gh/jscad%2Fcsg.js.svg)](https://badge.fury.io/gh/jscad%2Fcsg.js)
 [![Build Status](https://travis-ci.org/jscad/csg.js.svg)](https://travis-ci.org/jscad/csg.js)
 
-> Solid modeling library (2d & 3d)
```
Comment (kaosat-dev) [inline README.md]: A one liner short description is still needed I believe

File: README.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+## Overview
 
-
-Extraction of the csg.js lib from https://github.com/joostn/OpenJsCad, being itself a fork of https://github.com/evanw/csg.js
+Constructive Solid Geometry (CSG) is a modeling technique that uses Boolean operations like union and intersection to combine 3D solids. This library implements CSG operations on meshes elegantly and concisely using BSP trees, and is meant to serve as an easily understandable implementation of the algorithm.
```
Comment (kaosat-dev) [inline README.md]: Nice !

File: csg.js [hunk 3]
```diff
@@ -88,20 +88,31 @@ for solid CAD anyway.
...
+    CSG.defaultResolution2D = 32; // FIXME this seems excessive
+    /** Number of polygons per 360 degree revolution for 3D objects.
+     * @default
+     */
     CSG.defaultResolution3D = 12;
 
-    // Construct a CSG solid from a list of `CSG.Polygon` instances.
+    /** Construct a CSG solid from a list of `CSG.Polygon` instances.
+     * @param {CSG.Polygon[]} polygons - list of polygons
+     * @returns {CSG} new CSG object
```
Comment (kaosat-dev) [inline csg.js]: Small question : does JSDOC reckognize the {CSG} object as it is ? or does it need to be '{object}' : haven't tried this before.

File: csg.js [hunk 4]
```diff
@@ -1469,26 +1490,27 @@ for solid CAD anyway.
...
     CSG.cube = function(options) {
         var c, r;
         options = options || {};
         if (('corner1' in options) || ('corner2' in options)) {
             if (('center' in options) || ('radius' in options)) {
                 throw new Error("cube: should either give a radius and center parameter, or a corner1 and corner2 parameter")
             }
-            corner1 = CSG.parseOptionAs3DVector(options, "corner1", [0, 0, 0]);
-            corner2 = CSG.parseOptionAs3DVector(options, "corner2", [1, 1, 1]);
+            var corner1 = CSG.parseOptionAs3DVector(options, "corner1", [0, 0, 0]);
```
Comment (kaosat-dev) [inline csg.js]: good fix :)

File: csg.js [hunk 5]
```diff
@@ -1622,22 +1645,22 @@ for solid CAD anyway.
...
-    //
-    //     var cylinder = CSG.cylinder({
-    //       start: [0, -1, 0],
-    //       end: [0, 1, 0],
-    //       radius: 1,
-    //       resolution: 16
-    //     });
+    /** Construct a solid cylinder.
+     * @param {Object} [options] - options for construction
+     * @param {Vector} [options.start=[0,-1,0]] - start point of cylinder
```
Comment (kaosat-dev) [inline csg.js]: Is JSDOC picking up the 'sub options' correctly for this ?

File: csg.js [hunk 6]
```diff
@@ -2007,16 +2033,17 @@ for solid CAD anyway.
...
-    //
-    //     new CSG.Vector3D(1, 2, 3);
-    //     new CSG.Vector3D([1, 2, 3]);
-    //     new CSG.Vector3D({ x: 1, y: 2, z: 3 });
-    //     new CSG.Vector3D(1, 2); // assumes z=0
-    //     new CSG.Vector3D([1, 2]); // assumes z=0
+    /** Class Vector3D
+     * Represents a 3D vector with X, Y, Z coordinates.
+     * @constructor
+     *
```
Comment (kaosat-dev) [inline csg.js]: missing params ?

File: csg.js [hunk 7]
```diff
@@ -5738,12 +5780,24 @@ for solid CAD anyway.
         return CAG.fromPoints(points);
     };
 
-    //     var r = CSG.roundedRectangle({
-    //       center: [0, 0],
-    //       radius: [2, 1],
-    //       roundradius: 0.2,
-    //       resolution: 8,
-    //     });
+    /** Construct a rounded rectangle.
```
Comment (kaosat-dev) [inline csg.js]: nice !

File: csg.js (hunk 6 above)
Comment (z3dev) [inline csg.js]: Yep. But it's a start.

File: csg.js (hunk 5 above)
Comment (z3dev) [inline csg.js]: These are correct. The documentation generated looks really nice.

File: csg.js (hunk 3 above)
Comment (z3dev) [inline csg.js]: Not sure, but I believe it requires an array of CSG.Polygon.

File: README.md (hunk 1 above)
Comment (z3dev) [inline README.md]: I saw that in the "standard" but couldn't determine where to put the one line. Feel free to change.

**Comment by z3dev** [comment]
@kaosat-dev Thanks for the review and questions. I think this is a good start. Please merge so others can get started. Cheers

**Pull request merged**